src/views/activity.py

```diff
@@ -0,0 +1,49 @@
+
+from flask import Blueprint, render_template, request, flash, redirect, url_for, make_response, g
+from src.models.activity import Activity
+from src.models.contact import Contact
+from utility.decor import login_required
+
+
+phonebook = Contact()
+activities = Activity()
+
+activity = Blueprint('activity', __name__)
+
+@activity.route('/<int:contact_id>', methods=["GET"])
+@login_required
+def get_all(contact_id):
+
+    contact_name = phonebook.get_by_id(contact_id)['name']
+
+    rows = activities.get_all(contact_id)
+
+    return render_template('activity.html',
+                           rows=rows,
+                           contact_name=contact_name,
+                           contact_id=contact_id,
+                           username=g.user['client_name'])
+
+
+@activity.route('/<int:contact_id>', methods=["POST"])
+@login_required
+def add(contact_id):
+
+    action = request.form['action']
+    description = request.form['description']
+    date = request.form['date']
+    time = request.form['time']
+
+    activities.add(action, description, date, time,
+                   g.user['id'], contact_id)
+
+    return redirect(url_for('activity.get_all', contact_id=contact_id))
+
+
+@activity.route('/<int:contact_id>/delete:<int:activity_id>', methods=["POST"])
```

Comment:
This route should ideally be `DELETE /contacts/:id/activities/:activity`, or at least `POST /contacts/:id/activities/:activity/delete`.

Reply:
I have a prefix set in each blueprints. It gonna handle the "contacts" in the route. I handled the rest manually.

Reply:
Oh, Hang on! we are in the activity views and its prefix is "activity" not the "contacts".
I think I have to let prefixes go ...